src/ResourceManager/EventHub/Commands.EventHub/Cmdlets/AuthorizationRule/NewAzureEventhubAuthorizationRules.cs

```diff
@@ -0,0 +1,90 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.EventHub.Models;
+using Microsoft.Azure.Management.EventHub.Models;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Commands.EventHub.Commands
+{
+    /// <summary>
+    /// 'New-AzureRmRelayAuthorizationRule' Cmdlet creates a new AuthorizationRule
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, EventHubAuthorizationRuleVerb, DefaultParameterSetName = NamespaceAuthoRuleParameterSet, SupportsShouldProcess = true), OutputType(typeof(SharedAccessAuthorizationRuleAttributes))]
+    public class NewAzureRelayAuthorizationRule : AzureEventHubsCmdletBase
+    {
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 0,
+            HelpMessage = "Resource Group Name.")]
+        [ValidateNotNullOrEmpty]
+         public string ResourceGroupName { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 1, ParameterSetName = NamespaceAuthoRuleParameterSet,
+            HelpMessage = "Namespace Name.")][Parameter(ParameterSetName = EventhubAuthoRuleParameterSet)]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasNamespaceName)]
+        public string Namespace { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 2, ParameterSetName = EventhubAuthoRuleParameterSet,
+            HelpMessage = "EventHub Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasEventHubName)]
+        public string EventHub { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            Position = 3,
+            HelpMessage = "AuthorizationRule Name.")]
+        [ValidateNotNullOrEmpty]
+        [Alias(AliasAuthorizationRuleName)]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Rights, e.g.  \"Listen\",\"Send\",\"Manage\"")]
+        [ValidateNotNullOrEmpty]
+        public string[] Rights { get; set; }
```

Comment:
@v-Ajnava consider using `ValidateSet` if the values that you can provide for this parameter are restricted to a set of strings